lamindb/curators/__init__.py

```diff
@@ -3336,22 +3262,27 @@ def save_artifact(
     from ..models.artifact import add_labels, data_is_anndata, data_is_mudata
 
     if artifact is None:
-        if data_is_anndata(data):
-            artifact = Artifact.from_anndata(
+        if isinstance(data, pd.DataFrame):
+            artifact = Artifact.from_df(
                 data, description=description, key=key, revises=revises, run=run
             )
-        elif isinstance(data, pd.DataFrame):
-            artifact = Artifact.from_df(
+        elif data_is_anndata(data):
```

Comment:
Here should be `isinstance(data, ad.AnnData)` because paths are no longer possible at this stage.